### ISSUE TITLE
Why does FieldArray have to iterate all fields on every keypress?

### ISSUE THREAD

**Issue by JakeLin1024** [erikras/redux-form#1578]
Take the sample code as an example(slightly modified)

```
const renderMembers = ({ fields }) => {
  return (<ul>
    <li>
      <button type="button" onClick={() => fields.push({})}>Add Member</button>
    </li>
    {fields.map((member, index) =>{ 
      console.log("It's iterating members",index)  <==== here
      return (
        <li key={index}>
        <button
          type="button"
          title="Remove Member"
          onClick={() => fields.remove(index)}/>
        <h4>Member #{index + 1}</h4>
        <Field
          name={`${member}.firstName`}
          type="text"
          component={renderField}
          label="First Name"/>
        <Field
          name={`${member}.lastName`}
          type="text"
          component={renderField}
          label="Last Name"/>
        <FieldArray name={`${member}.hobbies`} component={renderHobbies}/>
       </li>)
    }
    )}
  </ul>
)}
```

lets say we have two members. Once you type the first name, console is like:
```
It's iterating members,0
It's iterating members,1
```
Since I dont know that much of the codebase, could you explain it a little bit?

**Comment by erikras** [comment]
Excellent catch! This was a bug. Fixed by #1590. I will comment here again when the fix is released.

**Comment by erikras** [comment]
Published in [`v6.0.1`](https://github.com/erikras/redux-form/releases/tag/v6.0.1).
